sdk/nodejs/runtime/settings.ts

```diff
@@ -169,11 +169,13 @@ export function setMockOptions(
         stack || opts.stack || "stack",
         opts.parallel || -1,
         opts.engineAddr || "",
-        opts.monitorAddr || "",
+        "mock",
```

Comment:
Do we still need to do this? I feel like there's something a bit suspect about the "mock" check in `getMonitor`, like why doesn't that just pick up from settings like `hasMonitor`?

Reply:
I'm not entirely sure why `getMonitor` checks if `getLocalStore` is undefined first before falling back to the global settings.  That was introduced in https://github.com/pulumi/pulumi/commit/5cfad4a909eb6162b57c770a9174340ec823bf97#diff-255dc1ec5266b9ecb936f7a31ab48f0bd1374a54eec3a3728f74f0de743358d9R225, so unfortunately there's not much explanation in that history.

I tried naively updating `getMonitor`, so it always just uses the globalStore, which is what I think we'd need to do here to make this work without this hack, but that doesn't work. So I think we still need this, but I'm open to ideas on how to make this work.

Reply:
Let's go with this for now. Maybe we'll get a brainwave on how to clean it up later.